### PR TITLE
Add cff

### PR DESCRIPTION
**Pull request by c-martinez**
I'm using this PR within my repo to have a code review before pushing upstream. 

As far as I understand, the workflow should be that you create a package, and add a CFF as follows:

```R
usethis::create_package('.')
usethis::use_cff()
```
And the output should be something like this:
```output
#> ✔ Writing 'citation.cff'.
# ☐ Edit 'citation.cff'. (see <https://book.the-turing-way.org/communication/citable/citable-cffinit.html> and <https://citation-file-format.github.io/cff-initializer-javascript/#/>)
```

Things I am not very sure about:
- [x] Can we fill some information in the created CITATION.cff file? It looks to me as if some fields could be pre-filled for the user, for example license, author name, release date?
- [x] Are tests sufficient?
- [ ] Is documentation sufficient?
- [x] Do we need to generate documentation via `roxygen2` somehow? (see I saw `man/*.Rd` files mention not to edit by hand, but I have no idea how this works).

### CODE REVIEW

**Comment by c-martinez** [comment]
Invited @PabRod as collaborator, once he accepts, I can assign him as reviewer.

**Comment by PabRod** [comment]
I booked some time to take a look at this on Tue 30th.

File: NAMESPACE [hunk 1]
```diff
@@ -106,6 +106,7 @@ export(use_cc0_license)
 export(use_ccby_license)
 export(use_circleci)
 export(use_circleci_badge)
+export(use_cff)
```
Comment (PabRod) [inline NAMESPACE]: Very good. Are you sure this is the first time you use R?

File: R/cff.R
Comment (PabRod) [inline R/cff.R:1]: I like the parallel with `use-citation()` (available in `R/citation.R`). It looks good, but I am slightly confused about their not-so-straightforward workflow.

**Comment by PabRod** [comment]
@c-martinez: 

> * Do we need to generate documentation via `roxygen2` somehow? (see I saw `man/*.Rd` files mention not to edit by hand, but I have no idea how this works).

Documentation can be generated from the `Build/More` menu. In R jargon, this is know as _roxygenizing_. 

It is usually a good idea to set it to roxygenize on every `Install and restart` from the `Build/More/Configure build tools/Documentation with Roxygen/Configure...` submenu.

Forgetting to roxygenize is a very common mistake. Indeed, there were functions in main that were not roxygenized.

**Comment by PabRod** [comment]
> * Can we fill some information in the created CITATION.cff file? It looks to me as if some fields could be pre-filled for the user, for example license, author name, release date?

I suggest keeping it as it is. I like how it mirrors `use_citation()`. If you disagree, please mark it again as _to do_.

**Comment by PabRod** [comment]
> * Are tests sufficient?

I'd say yes, ~~but for some reason yours is not running. The whole battery of tests shows many fails, and it stops before reaching this one. I couldn't manage to run this test only, and I don't fully understand why. My guess: it is related with their workflow.~~

Update: I was wrong. After updating my R to the latest version, only one test fails. It's not the cff one!

**Comment by PabRod** [comment]
> * Is documentation sufficient?

It is definitely not insufficient, but let's:

- [x] Draft a pull request explaining why `cff` is important
   - [ ] and consider editing the in-code documentation afterwards

**Comment by c-martinez** [comment]
I didn't see you edited your comment about tests. It looks like you can run a single test: https://testthat.r-lib.org/reference/test_file.html but I somehow don't get that to work (clearly need to learn how to use R studio properly).

Anyway, if you are satisfied with this review, I will merge and draft a PR.

**Comment by c-martinez** [comment]
I've created a PR, now let see if they have any feedback. Thanks for your help!

**Pull request merged**